kloppy/infra/serializers/tracking/signality.py

```diff
@@ -0,0 +1,241 @@
+                )
+            )
+
+        return periods
+
+    def deserialize(self, inputs: SignalityInputs) -> TrackingDataset:
+        metadata = json.load(inputs.meta_data)
+        venue_information = json.load(inputs.venue_information)
+        raw_data_feeds = []
+        for input_raw_data_feed in inputs.raw_data_feeds:
+            with open_as_file(input_raw_data_feed) as raw_data_feed_fp:
+                raw_data_feed = json.load(raw_data_feed_fp)
+                raw_data_feeds.append(raw_data_feed)
+        p1_raw_data = raw_data_feeds[0]
+
+        with performance_logging("Loading metadata", logger=logger):
+            frame_rate = self.__get_frame_rate(p1_raw_data)
+            teams = self.__create_teams(metadata)
+            periods = self.__get_periods(raw_data_feeds)
+            pitch_size_length = venue_information["pitch_size"][0]
+            pitch_size_width = venue_information["pitch_size"][1]
+
+        transformer = self.get_transformer(
+            pitch_length=pitch_size_length, pitch_width=pitch_size_width
+        )
+
+        with performance_logging("Loading data", logger=logger):
+            frames = []
+            for period, p_raw_data in zip(periods, raw_data_feeds):
+                n = 0
+                sample = 1.0 / self.sample_rate
+                if period.id == 1:
+                    frame_id_offset = 0
+                elif period.id == 2:
+                    frame_id_offset = p1_raw_data[-1]["idx"] + 1
+                else:
+                    raise ValueError(f"Period ID {period.id} not supported")
+                for raw_frame in p_raw_data:
+                    if n % sample == 0:
+                        frame = self._get_frame_data(
+                            teams, period, raw_frame, frame_id_offset
+                        )
+                        if frame.players_data and frame.timestamp > timedelta(
+                            seconds=0
+                        ):
+                            frame = transformer.transform_frame(frame)
+                            frames.append(frame)
+                    n += 1
+
+        attacking_directions = attacking_directions_from_multi_frames(
+            frames, periods
+        )
+        if attacking_directions[1] == AttackingDirection.LTR:
+            orientation = Orientation.HOME_AWAY
+        elif attacking_directions[1] == AttackingDirection.RTL:
+            orientation = Orientation.AWAY_HOME
+        else:
+            warnings.warn(
+                "Could not determine orientation of dataset, defaulting to NOT_SET"
+            )
+            orientation = Orientation.NOT_SET
+
+        metadata = Metadata(
+            teams=teams,
+            periods=periods,
+            frame_rate=frame_rate,
+            orientation=orientation,
+            pitch_dimensions=transformer.get_to_coordinate_system().pitch_dimensions,
+            coordinate_system=transformer.get_to_coordinate_system(),
+            flags=None,
```

Comment:
The data has "state" and "ball.team". I would assume these define the ball state and ball-owning team?